code/modules/clothing/shoes/miscellaneous.dm

```diff
@@ -410,6 +410,33 @@
 	item_state = "kneesock"
 	species_fit = list(INSECT_SHAPED)
 
+/obj/item/clothing/shoes/kneesocks/attackby(obj/item/weapon/W, mob/user)
+	..()
+	if(istype(W, /obj/item/weapon/soap))
+		to_chat(user, "<span class='notice'>You place \the [W] into \the [src].</span>")
+		if(do_after(user, src, 1 SECONDS))
+			user.drop_item(src)
+			if(!user.drop_item(W))
+				to_chat(user, "<span class='warning'>You can't let go of \the [W].</span>")
+				return
+			var/obj/item/weapon/soap_sock/I = new (get_turf(user))
+			user.put_in_hands(I)
+			W.transfer_fingerprints_to(I)
+			I.base_soap = W
+			W.forceMove(null)
+			qdel(src)
+	else if(istype(W, /obj/item/stack/sheet/mineral/brick))
+		var/obj/item/stack/sheet/mineral/brick/S = W
+		to_chat(user, "<span class='notice'>You place a brick into \the [src].</span>")
+		if(do_after(user, src, 1 SECONDS))
+			if(!user.drop_item(src))
+				to_chat(user, "<span class='warning'>You can't let go of \the [W].</span>")
+				return
+			var/obj/item/weapon/brick_sock/I = new (get_turf(user))
+			S.use(1)
```

Comment:
This is exploitable. You're supposed to check the return value of `use`: if it's falsy it means the stack doesn't have enough bricks left in it, so the proc should return. By not checking it, one could create a bunch of `brick_socks` with a single brick by quickly hitting multiple socks with it.

Reply:
Fixed in latest commit

Reply:
No, you didn't. Why is this call here? You're calling `use` one line below.

Reply:
I've compiled and tested it myself. No amount of spamming a stack of bricks of any size will create numerous copies of this with the current code. Does the use statement called within the if() also use a brick? The way I understood it, that just checks to see if a brick was, in fact, used (I may not understand it at all).

Reply:
Yes, you fixed the original issue with `if(!S.use(1))`, which makes this line obsolete. If you leave this line in, you will consume one brick, then try to consume another with `if(!S.use(1))`

Reply:
Ah, understood. Addressed in latest commit.